File: problem-005/problem-5.cpp
#include <stdio.h>


int main(void)
{
  const int MAX_DIVISOR = 20;
  int number = MAX_DIVISOR;
  bool flag = false;

  do
  {
    int a;
    for (a = 2; a <= MAX_DIVISOR; a++)
    {
      if (number % a != 0)
        break;
    }

    if (a > MAX_DIVISOR)
      break;

    number += 2;
  }
  while (true);

  printf("%d\n", number);
  return 0;
}

File: problem-005/problem-5.rb

PRIME_MULTIPLE = 9699690
nonprimes = [20,18,16,15,14,12]

number = PRIME_MULTIPLE


begin
  flag = true
  nonprimes.each do |x|
    if number % x != 0
      flag = false
      break
    end
  end
  break if flag
  number += PRIME_MULTIPLE
end until false

puts number
